dandiapi/api/tests/test_dandiset.py

```diff
@@ -139,6 +139,71 @@ def test_dandiset_rest_create(api_client, user):
     }
 
 
+@pytest.mark.django_db
+def test_dandiset_rest_create_with_identifier(api_client, user):
+    api_client.force_authenticate(user=user)
+    name = 'Test Dandiset'
+    identifier = '123456'
+    metadata = {'foo': 'bar', 'identifier': f'DANDI:{identifier}'}
+
+    response = api_client.post(
+        '/api/dandisets/',
+        {'name': name, 'metadata': metadata},
+        format='json',
+    )
```

Comment:
If we decide to change how `identifier` is specified, this (along with the other test) would probably become:

```
    identifier = '123456'
    metadata = {'foo': 'bar'}

    response = api_client.post(
        '/api/dandisets/',
        {'name': name, 'identifier': identifier, 'metadata': metadata},
        format='json',
    )
```